readthedocs/search/tests/test_views.py

```diff
@@ -67,26 +67,44 @@ def test_search_project_filter_language(self, client, project):
 
 @pytest.mark.django_db
 @pytest.mark.search
-class TestElasticSearch(object):
+class TestPageSearch(object):
     url = reverse('search')
 
     def _get_search_result(self, url, client, search_params):
         resp = client.get(url, search_params)
         assert resp.status_code == 200
 
         page = pq(resp.content)
-        result = page.find('.module-list-wrapper .module-item-title')
+        result = page.find('.module-list-wrapper .module-item')
         return result, page
 
     @pytest.mark.parametrize('data_type', ['content', 'headers', 'title'])
     @pytest.mark.parametrize('page_num', [0, 1])
-    def test_search_by_file_content(self, client, project, data_type, page_num):
+    def test_file_search(self, client, project, data_type, page_num):
         query = get_search_query_from_project_file(project_slug=project.slug, page_num=page_num,
                                                    data_type=data_type)
 
         result, _ = self._get_search_result(url=self.url, client=client,
                                             search_params={'q': query, 'type': 'file'})
-        assert len(result) == 1, ("failed"+ query)
+        assert len(result) == 1
+        assert query in result.text()
+
+    @pytest.mark.parametrize('case', ['upper', 'lower', 'title'])
+    def test_file_search_case_insensitive(self, client, project, case):
+        """Check File search is case insensitive
+
+        It tests with uppercase, lowercase and camelcase
+        """
+        query = get_search_query_from_project_file(project_slug=project.slug)\
+
+        cased_query = getattr(query, case)
```

Comment:
This feels really magical. I think there is a better way to write this code so that it's more obvious what it is doing. Perhaps by calling the function reference that is returned here, instead of just passing it in below.

Reply:
>Perhaps by calling the function reference that is returned here, instead of just passing it in below.

Do you mean something like
```python
if case == 'lower':
    query = query.lower()
```